tests/unit/system/classes/ImageResizerTest.php

```diff
@@ -4,68 +4,187 @@
+                'quality' => 73,
+                'extension' => 'jpg'
             ],
-            'fileInstance' => [
-                'source' => FileModel::first(),
-                'target' => '',
+        ], $imageResizer->getConfig());
+
+        // Resize with an customised defaults
+        Event::listen('system.resizer.getDefaultOptions', function (&$options) {
+            $options = array_merge($options, [
+                'mode' => 'fit',
+                'offset' => [2, 2],
+                'sharpen' => 23,
+                'interlace' => true,
+                'quality' => 73,
+            ]);
+        });
+
+        $imageResizer = new ImageResizer(
+            (new CmsController())->themeUrl('assets/images/october.png'),
+            100,
+            100,
+            [],
+        );
+        self::assertArraySubset([
+            'width' => 100,
+            'height' => 100,
+            'options' => [
+                'mode' => 'fit',
+                'offset' => [2, 2],
+                'sharpen' => 23,
+                'interlace' => true,
+                'quality' => 73,
+                'extension' => 'png',
             ],
-        ];
+        ], $imageResizer->getConfig());
+
+        Event::forget('system.resizer.getDefaultOptions');
+    }
+
+    public function testSources()
+    {
+        // Media URL
+        $this->setUpStorage();
+        $this->copyMedia();
+
+        $imageResizer = new ImageResizer(
+            MediaLibrary::url('october.png'),
+            100,
+            100,
+        );
+        $this->assertEquals('png', $imageResizer->getConfig()['options']['extension']);
+
+        // Plugin URL (also tests absolute URLs)
+        $imageResizer = new ImageResizer(
+            URL::to('plugins/database/tester/assets/images/avatar.png'),
+            100,
+            100,
+        );
+        $this->assertEquals('png', $imageResizer->getConfig()['options']['extension']);
+
+        // Relative URL
+        $imageResizer = new ImageResizer(
+            '/plugins/database/tester/assets/images/avatar.png',
+            100,
+            100,
+        );
+        $this->assertEquals('png', $imageResizer->getConfig()['options']['extension']);
+
+        // Module URL
+        $imageResizer = new ImageResizer(
+            Backend::skinAsset('assets/images/favicon.png'),
+            100,
+            100,
+        );
+        $this->assertEquals('png', $imageResizer->getConfig()['options']['extension']);
+
+        // Still to test - FileModel URL and FileModel instance
     }
 
     public function testInvalidInput()
     {
+        $this->markTestIncomplete();
+
         $providedPath = '/plugins/october/demo/assets/NOTPRESENT.png';
     }
+
+    protected function setUpStorage()
+    {
+        $this->app->useStoragePath(base_path('storage/temp'));
+
+        config(['filesystems.disks.test_local' => [
+            'driver' => 'local',
+            'root'   => storage_path('app'),
```

Comment:
Shouldn't this be `storage_path('temp/app');`

Reply:
@LukeTowers nope - the call above sets the storage path to `temp` already - see `$this->app->useStoragePath` :).